src/context/AuthContext.tsx

```diff
@@ -0,0 +1,87 @@
+'use client';
+
+import React, {
+  createContext,
+  useState,
+  useContext,
+  useEffect,
+  ReactNode,
+} from 'react';
+
+import { getUser } from '../client';
+import { User } from '../types';
+
+type AuthContextProps = {
+  user: User | null;
+  login: (token: string) => void;
+  logout: () => void;
+};
+
+export const AuthContext = createContext<AuthContextProps | undefined>(
+  undefined,
+);
+
+type AuthProviderProps = {
+  children: ReactNode;
+};
+
+export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
That is exactly what I wrote above! Null as the default value here is actually better. I would go for undefined myself cause I am lazy. But this is better.

But like I said, we could have empty object as default for DX